### PR TITLE
Update ButtonLink hover underline

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

- Add underline to `ButtonLink` to add a better visual aid for users on hover. 
- The underline is 2px in thickness for sizes: Auto, Sm, Md, Lg with an offset of 4px. For size inherit, text-decoration will auto-adopt the font weight and position to be only 2px offset since these use cases will mostly be in paragraphs/etc. 
- Underline only goes under the text and not the icons

### Before

Screenshot 2023-07-12 at 5 15 13 PM

### After
Screenshot 2023-07-13 at 2 17 01 PM

Screenshot 2023-07-13 at 2 33 46 PM

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/button-link/button-link.scss [hunk 1]
```diff
@@ -1,5 +1,10 @@
-
 .mm-button-link {
+  &:hover:not(&--disabled) {
+    text-decoration: underline;
+    text-decoration-thickness: 2px;
+    text-underline-offset: 4px;
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-link/button-link.scss:5]: Forgive me if I missed this discussion but what's the reason to have different thicknesses across sizes? I was expecting it just to be the default across the board. I kind of like the distinction between button and link though

**Review by georgewrmarshall — APPROVED** [review]
Excellent. Just have once question

File: ui/components/component-library/button-link/button-link.scss (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/component-library/button-link/button-link.scss:5]: Ah right can you add the figma link or screenshot to the description I can't find it
Screenshot 2023-07-13 at 4 19 42 PM

**Comment by darkwing** [comment]
Yeah, I really like this!  Nice job!

**Review by darkwing — APPROVED** [review]

**Pull request merged**